src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -86,90 +108,185 @@ public DateTimeOffset EndTime
+        /// <returns>Sum value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public double GetHistogramSum()
+        {
+            if (this.aggType != AggregationType.Histogram && this.aggType != AggregationType.HistogramSumCount && this.aggType != AggregationType.HistogramWithMinMax)
             {
-                throw new NotSupportedException($"{nameof(this.GetGaugeLastValueDouble)} is not supported for this metric type.");
+                this.ThrowNotSupportedMetricTypeException(nameof(this.GetHistogramSum));
             }
+
+            return this.histogramBuckets.SnapshotSum;
         }
 
-        public long GetHistogramCount()
+        /// <summary>
+        /// Gets the minimum value of the histogram associated with the metric point.
+        /// </summary>
+        /// <remarks>
+        /// Applies to <see cref="MetricType.HistogramWithMinMax"/> metric type.
+        /// </remarks>
+        /// <returns>Minimum value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public double GetHistogramMin()
         {
-            if (this.aggType == AggregationType.Histogram || this.aggType == AggregationType.HistogramSumCount)
+            if (this.aggType != AggregationType.HistogramWithMinMax)
             {
-                return this.primaryValue.SnapshotAsLong;
+                this.ThrowNotSupportedMetricTypeException(nameof(this.GetHistogramMin));
             }
-            else
+
+            if (this.runningValue.AsLong == 0)
             {
-                throw new NotSupportedException($"{nameof(this.GetHistogramCount)} is not supported for this metric type.");
+                throw new InvalidOperationException("No values have been recorded yet.");
             }
+
+            return this.histogramBuckets.SnapshotMin;
         }
 
-        public double GetHistogramSum()
+        /// <summary>
+        /// Gets the maximum value of the histogram associated with the metric point.
+        /// </summary>
+        /// <remarks>
+        /// Applies to <see cref="MetricType.HistogramWithMinMax"/> metric type.
+        /// </remarks>
+        /// <returns>Maximum value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public double GetHistogramMax()
         {
-            if (this.aggType == AggregationType.Histogram || this.aggType == AggregationType.HistogramSumCount)
+            if (this.aggType != AggregationType.HistogramWithMinMax)
             {
-                return this.secondaryValue.SnapshotAsDouble;
+                this.ThrowNotSupportedMetricTypeException(nameof(this.GetHistogramMax));
             }
-            else
+
+            if (this.runningValue.AsLong == 0)
             {
-                throw new NotSupportedException($"{nameof(this.GetHistogramSum)} is not supported for this metric type.");
+                throw new InvalidOperationException("No values have been recorded yet.");
             }
```

Comment:
What should the behaviour be when calling GetHistogramMax() if no values have been recorded?